### DISCUSSION TITLE
Html form security

### DISCUSSION THREAD

**Discussion by nitins11** [ONDC-Official/ONDC-FIS-Specifications#154] [General]
The four forms from BPPs mentioned below are open URLs without any authentication info. How do we ensure any submission on these URLs are actually originating from a borrower-BAPs since these are open?

1. Form name , pan ,address
2. Form for loan amount
3. Form for kyc 
4. Form for account number

**Comment by Sarveshwaran20** [comment]
Ensuring the authenticity of submissions on open URLs without authentication can be challenging. To enhance security and verify the origin of submissions, consider implementing the following measures:

1. Captcha or reCAPTCHA: Integrate a captcha or reCAPTCHA mechanism to prevent automated submissions and ensure that submissions are coming from real users.

2. Session Tokens: Implement session tokens or cookies to track user sessions. This won't prevent manual submissions from non-borrowers, but it adds a layer of session-based security.

3. IP Address Logging: Log the IP addresses of users submitting forms. While this doesn't guarantee authentication, it can be useful for tracking and analysis.

4. User Authentication: If possible, require users to authenticate before accessing the forms. This could involve implementing a login system or using third-party authentication services.

5. Encryption: Ensure that the communication between the user and the server is encrypted using HTTPS to prevent eavesdropping and tampering.

6. Browser Fingerprinting: Utilize browser fingerprinting techniques to identify unique characteristics of the user's browser, adding another layer of identification.

Remember that none of these methods are foolproof on their own, and a combination of multiple strategies provides a more robust security posture. It's crucial to continuously monitor and update security measures to adapt to evolving threats.

**Comment by Sarveshwaran20** [comment]
Did it work
